eqs/src/routes.rs

```diff
@@ -148,19 +148,21 @@ pub async fn get_transaction(
         .cloned())
 }
 
-// pub async fn get_transaction_by_hash(
-//     bindings: &HashMap<String, RouteBinding>,
-//     query_result_state: &QueryResultState,
-// ) -> Result<Option<CommittedCapeTransition>, tide::Error> {
-//     if let Some(txn_id) = query_result_state
-//         .transaction_id_by_hash
-//         .get(&bindings[":hash"].value.to::<Commitment<CapeTransition>>()?)
-//     {
-//         Ok(query_result_state.transaction_by_id.get(txn_id).cloned())
-//     } else {
-//         Ok(None)
-//     }
-// }
+pub async fn get_transaction_by_hash(
+    bindings: &HashMap<String, RouteBinding>,
+    query_result_state: &QueryResultState,
+) -> Result<Option<CommittedCapeTransition>, tide::Error> {
+    if let Some(txn_id) = query_result_state.transaction_id_by_hash.get(
+        &bindings[":hash"]
+            .value
+            .to::<CommitmentToCapeTransition>()?
+            .0,
+    ) {
+        Ok(query_result_state.transaction_by_id.get(txn_id).cloned())
```

Comment:
Maybe it should be an error if the hash exists but its id is invalid? This violates internal consistency assumptions right?

Reply:
Making that an error case, but I'm wondering if I shouldn't also switch to forming the tide::Response directly for this endpoint, so that we can have a (non-error) `tide::Response::new(tide::StatusCode::NotFound)` for the not found case.

Reply:
Would it make sense for the net::server::response function to automate that for `Result<Option<BodyType>>`?

Reply:
I think different things make sense for different endpoints. In my pretty uninformed opinion, `tide::StatusCode::NotFound` feels more error-ish, like you asked for something you shouldn't have. Whereas for something that is explicitly optional, returning a `JSON` object that might be `null` seems easier to work with in JS, and that corresponds to returning `None` with status 200.